internal/provider/data_source_gitlab_project_tags.go

```diff
@@ -0,0 +1,154 @@
+				Required:    true,
+			},
+			"order_by": {
+				Description: "Return tags ordered by `name` or `updated` fields. Default is `updated`.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"sort": {
+				Description: "Return tags sorted in `asc` or `desc` order. Default is `desc`.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"search": {
+				Description: "Return list of tags matching the search criteria. You can use `^term` and `term$` to find tags that begin and end with `term` respectively. No other regular expressions are supported.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"tags": {
+				Description: "List of repository tags from a project.",
+				Type:        schema.TypeList,
+				Computed:    true,
+				Elem:        dataTagsSchema,
+			},
+		},
+	}
+})
+
+var dataTagsSchema = &schema.Resource{
+	Schema: map[string]*schema.Schema{
+		"commit": {
+			Description: "The commit associated with the tag ref.",
+			Type:        schema.TypeSet,
+			Computed:    true,
+			Set:         schema.HashResource(commitSchema),
+			Elem:        commitSchema,
+		},
+		"release": {
+			Description: "The release associated with the tag.",
+			Type:        schema.TypeSet,
+			Computed:    true,
+			Set:         schema.HashResource(releaseNoteSchema),
+			Elem:        releaseNoteSchema,
+		},
+		"name": {
+			Description: "The name of the tag.",
+			Type:        schema.TypeString,
+			Computed:    true,
+		},
+		"target": {
+			Description: "The unique id assigned to the commit by Gitlab.",
+			Type:        schema.TypeString,
+			Computed:    true,
+		},
+		"message": {
+			Description: "Creates annotated tag.",
+			Type:        schema.TypeString,
+			Computed:    true,
+		},
+		"protected": {
+			Description: "Bool, true if tag has tag protection.",
+			Type:        schema.TypeBool,
+			Computed:    true,
+		},
+	},
+}
+
+func dataSourceGitlabProjectTagsRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*gitlab.Client)
+
+	project := d.Get("project").(string)
+	options := gitlab.ListTagsOptions{
+		ListOptions: gitlab.ListOptions{
+			PerPage: 20,
+			Page:    1,
+		},
+	}
+
+	if v, ok := d.GetOk("order_by"); ok {
+		options.OrderBy = gitlab.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("sort"); ok {
+		options.Sort = gitlab.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("search"); ok {
+		options.Search = gitlab.String(v.(string))
+	}
+
+	var tags []*gitlab.Tag
+	for options.Page != 0 {
+		paginatedTags, resp, err := client.Tags.ListTags(project, &options)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		tags = append(tags, paginatedTags...)
+		options.Page = resp.NextPage
+	}
+
+	optionsHash, err := hashstructure.Hash(&options, nil)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	log.Printf("[DEBUG] get gitlab tags from project: %s", project)
+	d.SetId(fmt.Sprintf("%s-%d", project, optionsHash))
```

Comment:
```suggestion
	d.SetId(fmt.Sprintf("%s:%d", project, optionsHash))
```

it's quite common to use a colon to separate id components :)

Reply:
Without problem) I'll change this code tomorrow.